src/wui/fieldaction.cc

```diff
@@ -1223,9 +1223,22 @@ void show_field_action(InteractiveBase* const ibase,
 
 class ShipSelectionWindow : public UI::UniqueWindow {
 	static constexpr int kPadding = 4;
-	static constexpr int kListWidth = 200;
 	static constexpr int kListHeight = 200;
 
+	static inline std::string ship_description(const Widelands::Ship* ship) {
+		std::string status;
+		if (ship->get_ship_type() == Widelands::ShipType::kWarship) {
+			status = format(
+			/** TRANSLATORS: Placeholders are the current and maximum health of the warship. */
+			pgettext("ship_state", "Warship, %1$u / %2$u"), ship->get_hitpoints(), ship->descr().max_hitpoints_);
```

Comment:
Divide hitpoints by 100?

And please show attack bonus as well, e.g. "Warship, 500/500, +30%". (to me it's part of the "health" bar)